deadlines-api/src/main/java/com/palantir/deadlines/api/DeadlineExpiredException.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * (c) Copyright 2025 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.deadlines.api;
+
+import com.palantir.logsafe.Arg;
+import com.palantir.logsafe.Safe;
+import com.palantir.logsafe.SafeLoggable;
+import java.util.List;
+
+/**
+ * Indicates that a deadline has expired.
+ */
+public abstract class DeadlineExpiredException extends RuntimeException {
```

Comment:
perhaps we should split this class out into a separate PR so that we can publish it in the API for this library and allow consumers to adopt logic that handles these exception types before we merge a change to actually throw the exception

Reply:
Makes sense to me!

can we make this a sealed type? We should also implement SafeLoggable imo

Reply:
is there much utility in having a base type at all here?

Reply:
see https://github.com/palantir/deadlines-java/pull/25